EpicLoot/Adventure/Feature/Bounties.cs

```diff
@@ -27,12 +27,46 @@ public List<BountyInfo> GetAvailableBounties()
             return GetAvailableBounties(GetCurrentInterval());
         }
 
+        public bool BossBountiesGated()
+        {
+            switch (EpicLoot.BossBountyMode.Value) 
+            {
+                case GatedBountyMode.BossKillUnlocksCurrentBiomeBounties:
+                case GatedBountyMode.BossKillUnlocksNextBiomeBounties:
+                    return true;
+                default:
+                    return false;
+            }
+        }
+
         public List<BountyInfo> GetAvailableBounties(int interval, bool removeAcceptedBounties = true)
         {
             var player = Player.m_localPlayer;
             var random = GetRandomForInterval(interval, RefreshInterval);
 
             var bountiesPerBiome = new MultiValueDictionary<Heightmap.Biome, BountyTargetConfig>();
+            
+            var defeatedBossBiomes = new List<Heightmap.Biome>();
+            var biomeBosses = AdventureDataManager.Config.Bounties.Bosses;
+            var previousBossKilled = false;
+
+            if (BossBountiesGated())
+            {
+                foreach (var bossConfig in AdventureDataManager.Config.Bounties.Bosses)
+                {
+                    if (ZoneSystem.instance.GetGlobalKey($"defeated_{bossConfig.BossName}"))
```

Comment:
So instead of doing this string construction here, it would be better to include the whole defeated_whatever key in the config. That way, if they change the format for some reason, then we could just update the config.